lib/spack/spack/modules/lmod.py

```diff
@@ -223,15 +223,18 @@ class LmodFileLayout(BaseFileLayout):
     @property
     def arch_dirname(self):
         """Returns the root folder for THIS architecture"""
-        arch_folder = '-'.join([
-            str(self.spec.platform),
-            str(self.spec.os),
-            str(self.spec.target.family)
-        ])
-        return os.path.join(
-            self.dirname(),  # root for lmod module files
-            arch_folder,  # architecture relative path
-        )
+        # Architecture sub-folder
+        arch_folder_conf = spack.config.get(
+            'modules:%s:arch_folder' % self.conf.name, True)
```

Comment:
I wonder if we can completely drop this method and avoid having a new config attribute in `modules.yaml`. The idea would be to just use the `root` folder under:
```yaml
modules:
  default:
    root:
      tcl: <root-folder>
```
and complement that with a suitable value for the `projections` attribute to get the same default as we have now. I think an implementation like that would me more consistent in the sense that anything that is a function of the spec in the modulefile name would be delegated to projections. Would that be easy to implement in your opinion? Am I overlooking some corner cases?